src/table/context/useContextSelector.ts

```diff
@@ -1,35 +1,28 @@
-import { Context, useContext, useEffect, useMemo, useReducer, useRef } from 'react';
+import { Context, useContext, useEffect, useState } from 'react';
+import useMutableProp from '../virtualized-table/hooks/use-mutable-prop';
 import { getSelectorContext, SelectorContextType } from './createSelectorProvider';
 
-type Selector<TContext, TSelected> = (state: TContext) => TSelected;
+export type Selector<TContext, TSelected> = (state: TContext) => TSelected;
 
 export function useContextSelector<T, TSelected>(context: Context<T>, selector: Selector<T, TSelected>): TSelected {
   const accessorContext = getSelectorContext(context) as Context<SelectorContextType<T>>;
   const [accessor, listeners] = useContext(accessorContext);
-  const [, forceUpdate] = useReducer((dummy) => dummy + 1, 0);
-
-  const latestSelector = useRef(selector);
-  const latestSelectedState = useRef<any>();
-
   const currentValue = accessor();
 
   if (currentValue === undefined) {
     throw new Error('You must call useContextSelector inside a valid context.');
   }
 
-  const selectedState = useMemo(() => selector(currentValue), [currentValue, selector]);
```

Comment:
`selector` is always a new value as we do not wrap the `selector` in a `useCallback` when creating it. Instead it's always re-create. Ex: `useContextSelector(TableContext, (value) => value.baseProps)`.

So the `selectedState` is resolved/update on each render.

Reply:
Don't think so, `currentValue` could still be a new value which would recreate `selectedState`. So the consumer of this hook could still get a new `selectedState` before the actual state update happens.